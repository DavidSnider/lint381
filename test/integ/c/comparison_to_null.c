if (foo == 0) {}
if (foo == '\0') {}
if (foo == NULL) {}
if (foo != 0) {}
if (foo != '\0') {}
if (foo != NULL) {}
if (foo) {}
if (foo == 10) {}
