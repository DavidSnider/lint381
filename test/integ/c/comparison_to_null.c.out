comparison_to_null.c:2:9: error: Comparison to '\0' should be avoided
if (foo == '\0') {}
        ^^^^^^^
comparison_to_null.c:3:9: error: Comparison to NULL should be avoided
if (foo == NULL) {}
        ^^^^^^^
comparison_to_null.c:5:9: error: Comparison to '\0' should be avoided
if (foo != '\0') {}
        ^^^^^^^
comparison_to_null.c:6:9: error: Comparison to NULL should be avoided
if (foo != NULL) {}
        ^^^^^^^
